src/Ocelot/DependencyInjection/Features.cs

```diff
@@ -16,10 +24,29 @@ public static class Features
     /// Read The Docs: <see href="https://ocelot.readthedocs.io/en/latest/features/ratelimiting.html">Rate Limiting</see>.
     /// </remarks>
     /// <param name="services">The services collection to add the feature to.</param>
+    /// <param name="configurationRoot">Root configuration object.</param>
     /// <returns>The same <see cref="IServiceCollection"/> object.</returns>
-    public static IServiceCollection AddRateLimiting(this IServiceCollection services) => services
-        .AddSingleton<IRateLimiting, RateLimiting.RateLimiting>()
-        .AddSingleton<IRateLimitStorage, MemoryCacheRateLimitStorage>();
+    public static IServiceCollection AddRateLimiting(this IServiceCollection services, IConfiguration configurationRoot)
```

Comment:
I will add a new method instead of overriding existing one. I assumed this method is for internal purposes, but now I realize that it was a wrong assumption given that this method is public.

Reply:
The decision hinges on the ultimate logic! Perhaps separating the internal extension methods is not essential.
I advise to revisit this once all the logic has been developed.